src/ArrayOfArraysView.hpp

```diff
@@ -720,9 +721,8 @@ class ArrayOfArraysView
     // RAJA::inclusive_scan fails on empty input range
     if( numSubArrays > 0 )
     {
-      // const_cast needed until for RAJA bug.
-      RAJA::inclusive_scan< POLICY >( const_cast< INDEX_TYPE * >( capacities ),
-                                      const_cast< INDEX_TYPE * >( capacities + numSubArrays ),
+      RAJA::inclusive_scan< POLICY >( capacities,
```

Comment:
This function should probably call `resizeFromOffsets` once the scan has been performed I think.

Reply:
So I couldn't just call it, because the result of the scan is written into `m_offsets`, but its old values are still needed in `resizeFromOffsets`, plus it would then perform a self-copy of `m_offsets`. I ended up pulling all the shared code into a private function `resizeFromOffsetsImpl`, which both `resizeFromOffsets` and `resizeFromCapacities` call, passing a callback that is invoked when new offsets must be populated.